### PR TITLE
Android: Implement new EmulationActivity.

### PR DESCRIPTION
**Pull request by sigmabeta**
Design wise, the biggest change is that the emulation is now contained within a Fragment. This doesn't yield any benefits that I know of yet, but will be helpful in making the transition animations work seamlessly. 

In terms of functionality, the two big gains are that a bug was fixed causing the activity not to end properly, and the activity now uses android's Immersive mode to render the emulation to the full screen, with the UI chrome showing only on demand and hiding after 3 seconds. 

As a result, the back button is freed up for its normal function of backing straight out of the activity. This would be less annoying when the saving/loading functionality is put in, but even so I'll be adding an "are you sure?" dialog.

### CODE REVIEW

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+import org.dolphinemu.dolphinemu.NativeLibrary;
+import org.dolphinemu.dolphinemu.R;
+import org.dolphinemu.dolphinemu.fragments.EmulationFragment;
+import org.dolphinemu.dolphinemu.settings.input.InputConfigFragment;
+
+import java.util.List;
+
+public final class EmulationActivity extends Activity
+{
+
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java]: unnecessary newline.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+		}
+	}
+
+	// Gets button presses
+	@Override
+	public boolean dispatchKeyEvent(KeyEvent event)
+	{
+		int action = 0;
+
+
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java]: one newline too many,

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/fragments/EmulationFragment.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	/**
+	 * Initialize anything that doesn't depend on the layout / views in here.
+	 */
+	@Override
+	public void onCreate(Bundle savedInstanceState)
+	{
+		super.onCreate(savedInstanceState);
+
+		mPreferences = PreferenceManager.getDefaultSharedPreferences(getActivity());
+
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/fragments/EmulationFragment.java]: unnecessary newline.

**Comment by sigmabeta** [comment]
@dolphin-emu-bot rebuild

**Comment by sigmabeta** [comment]
@lioncash This should be ready to go now. Buildbot was behaving as if I did a `git rm --cached` on those two model classes I got rid of (but I didn't, that doesn't make sense.) I can't find any way around the problem other than to add that extra commit.

In fact I bet Buildbot is still trying to compile those two files, but I made them blank, so at least it doesn't fail.

**Comment by Parlane** [comment]
@dolphin-emu-bot rebuild

**Comment by Parlane** [comment]
@dolphin-emu-bot rebuild

**Pull request merged**